js_test.go

```diff
@@ -1268,3 +1268,51 @@ func TestStreamNameBySubject(t *testing.T) {
 		}
 	}
 }
+
+func TestJetStreamTransform(t *testing.T) {
+	s := RunBasicJetStreamServer()
+	defer shutdownJSServerAndRemoveStorage(t, s)
+
+	nc, js := jsClient(t, s)
+	defer nc.Close()
+
+	_, err := js.AddStream(&StreamConfig{
+		Name:             "ORIGIN",
+		Subjects:         []string{"test"},
+		SubjectTransform: &SubjectTransformConfig{Source: ">", Destination: "transformed.>"},
+		Storage:          MemoryStorage,
+	})
+	if err != nil {
+		t.Fatalf("Unexpected error: %v", err)
+	}
+
+	err = nc.Publish("test", []byte("1"))
+	if err != nil {
+		t.Fatalf("Unexpected error: %v", err)
+	}
+
+	_, err = js.AddStream(&StreamConfig{
+		Subjects: []string{},
+		Name:     "SOURCING",
+		Sources:  []*StreamSource{{Name: "ORIGIN", FilterSubject: ">", SubjectTransformDest: "fromtest.>"}},
+		Storage:  MemoryStorage,
+	})
+	if err != nil {
+		t.Fatalf("Unexpected error: %v", err)
+	}
+
+	// Create a sync subscription with an in-memory ephemeral consumer.
+	sub, err := js.SubscribeSync(">", ConsumerMemoryStorage(), BindStream("SOURCING"))
```

Comment:
Maybe add what other subjects would work around here besides full wildcard? Or would only full wildcard work because it has to match the FilterSubject?

Reply:
Sure, I changed it to `fromtest.>` and changed the stream to not listen to any subjects. And you inadvertently make the point I made in the conversation above, with the removal of checking the subject is a subset of the stream's listen FilterSubject the test with no filter subject set in the stream (or set to something that doesn't intersect fromtest.>) now works when using `fromtest.>` in the SubscribeSync while before it would not have.

Reply:
Thanks for the explanation!